File: first_nonrepeated_char.swift
#!/usr/bin/env swift
// find first nonrepeated character in a string

func findFirstNonrepeatedCharacter(s: String) -> Character? {
  guard s.characters.count > 0 else { return nil }
  var table = [Character:Int]()
  for c in s.characters {
    table[c] = s.characters.filter { $0 == c }.count
  }
  for c in s.characters {
    if table[c] == 1 { return c }
  }
  return nil
}


for w in ["total", "teeter"] { print(findFirstNonrepeatedCharacter(w)) }

File: first_nonrepeated_char.py
#!/usr/bin/env python
# find first nonrepeated character in a string

def first_nonrepeated_char(s):
  characters_dictionary = {}
  for i in xrange(0,len(s)):
    if s[i] in characters_dictionary:
      characters_dictionary[s[i]] += 1
    else:
      characters_dictionary[s[i]] = 1
  for i in xrange(0,len(s)):
    if characters_dictionary[s[i]] == 1:
      return s[i]
  return None

for w in ["total","teeter"]:
  print first_nonrepeated_char(w)
